src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/AndroidBrokerProxy.cs

```diff
@@ -205,32 +205,51 @@ public AdalResultWrapper GetAuthTokenInBackground(AuthenticationRequest request,
         public Intent GetIntentForBrokerActivity(AuthenticationRequest request, Activity callerActivity)
         {
             Intent intent = null;
-            IAccountManagerFuture result = null;
+
             try
             {
+                IAccountManagerFuture result = null;
                 // Callback is not passed since it is making a blocking call to get
                 // intent. Activity needs to be launched from calling app
                 // to get the calling app's metadata if needed at BrokerActivity.
                 Bundle addAccountOptions = GetBrokerOptions(request);
+
+                _logger.InfoPii("BrokerProxy: Broker options:" +
+                    "\nAuthority: " + request.Authority +
+                    "\nBroker Account Name: " + request.BrokerAccountName,
+                    "BrokerProxy: Broker options: "  +
+                    "\nBroker Account Name: " + request.BrokerAccountName);
+
                 result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
                     BrokerConstants.AuthtokenType, null, addAccountOptions, null,
-                    null, new Handler(callerActivity.MainLooper));
+                    null, GetPreferredLooper(callerActivity));
+
+                if (result == null)
+                {
+                    _logger.Info("BrokerProxy: IAccountManagerFuture returned null. ");
```

Comment:
wondering if this should say that the android account manager AddAccount didn't return any results?

Reply:
good idea. updated.

---
In reply to: [288846301](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1598#discussion_r288846301) [](ancestors = 288846301)